processing/src/main/java/io/druid/segment/data/GenericIndexedWriter.java

```diff
@@ -82,6 +132,11 @@ public void write(T objectToWrite) throws IOException
     valuesOut.write(bytesToWrite);
 
     headerOut.write(Ints.toByteArray((int) valuesOut.getCount()));
+    headerOutLong.write(Longs.toByteArray(valuesOut.getCount()));
```

Comment:
But I can totally avoid writing into headerOut after the switch. Will update the pr not to write into headerOut after switch.